### ISSUE TITLE
[Feature request] Type declarations

### ISSUE THREAD

**Issue by danieltroger** [Sawtaytoes/reactjs-solidjs-bridge#4]
Would be amazing to have some typescript type declarations for this

**Comment by Sawtaytoes** [comment]
That's true. I wrote this long before I'd learned TypeScript. Feel free to add them! I haven't had time to work on this library, and using SolidJS at work is getting farther and farther away based on the tech stack at my current job.

**Comment by danieltroger** [comment]
I see. Sad that you can't use solid :( What are you using instead?

Going without type declarations right now but if I write some later I'll add them here.

**Comment by Sawtaytoes** [comment]
I'm using React only because that's what's established where I work.

**Comment by vladnicula** [comment]
would you accept/merge a complete "translation" to typescript? I don't promise anything, but if we decide to adopt your package I might have to do it for work anyway.

**Comment by divinespear** [comment]
You can try [vite-plugin-dts](https://www.npmjs.com/package/vite-plugin-dts) or [vite-dts](https://www.npmjs.com/package/vite-dts) while building with vite.

**Comment by divinespear** [comment]
You can start with command:
`tsc --declaration --rootDir src --outDir dist --noEmit false --emitDeclarationOnly --allowJs false --checkJs false`

many errors while running (is it expected maybe?) but `d.ts` files are made.

**Comment by Sawtaytoes** [comment]
I started fixing up the SolidJS side of things since I started a SolidJS project and need to render React components.

I converted most of the project to TypeScript in the `master` branch (forgot to make a feature branch). I also converted to Vite because it wasn't working right with the original Webpack implementation.

SolidJS is not playing nicely with the ReactJS though. Not sure how to get both to work. It's something to do with the Vite plugins. If I can fix that, I think it's in a "good enough" state.
